cmd/juju/charmhub/info_test.go

```diff
@@ -34,11 +36,97 @@ func (s *infoSuite) TestRun(c *gc.C) {
 	defer s.setUpMocks(c).Finish()
 	s.expectInfo()
 	command := &infoCommand{api: s.api, charmOrBundle: "test"}
+	cmdtesting.InitCommand(command, []string{})
 	ctx := commandContextForTest(c)
 	err := command.Run(ctx)
 	c.Assert(err, jc.ErrorIsNil)
 }
 
+func (s *infoSuite) TestRunJSON(c *gc.C) {
+	defer s.setUpMocks(c).Finish()
+	s.expectInfo()
+	command := &infoCommand{api: s.api, charmOrBundle: "test"}
+	cmdtesting.InitCommand(command, []string{"--format", "json"})
+	ctx := commandContextForTest(c)
+	err := command.Run(ctx)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(cmdtesting.Stdout(ctx), gc.Equals, `{"type":"charm","id":"charmCHARMcharmCHARMcharmCHARM01","name":"wordpress","description":"This will install and setup WordPress optimized to run in the cloud.","publisher":"Wordress Charmers","summary":"WordPress is a full featured web blogging tool, this charm deploys it.","series":["bionic","xenial"],"store-url":"https://someurl.com/wordpress","tags":["app","seven"],"charm":{"config":{"Options":{"agility-ratio":{"Type":"float","Description":"A number from 0 to 1 indicating agility.","Default":null},"outlook":{"Type":"string","Description":"No default outlook.","Default":null},"reticulate-splines":{"Type":"boolean","Description":"Whether to reticulate splines on launch, or not.","Default":null},"skill-level":{"Type":"int","Description":"A number indicating skill.","Default":null},"subtitle":{"Type":"string","Description":"An optional subtitle used for the application.","Default":""},"title":{"Type":"string","Description":"A descriptive title used for the application.","Default":"My Title"},"username":{"Type":"string","Description":"The name of the initial account (given admin permissions).","Default":"admin001"}}},"relations":{"provides":{"source":"dummy-token"},"requires":{"sink":"dummy-token"}},"subordinate":false,"used-by":["wordpress-everlast","wordpress-jorge","wordpress-site"]},"channel-map":{"latest/stable":{"released-at":"2019-12-16T19:44:44.076943+00:00","track":"latest","risk":"stable","revision":16,"size":12042240,"version":"1.0.3"}},"tracks":["latest"]}
+`)
+}
+
+func (s *infoSuite) TestRunYAML(c *gc.C) {
+	defer s.setUpMocks(c).Finish()
+	s.expectInfo()
+	command := &infoCommand{api: s.api, charmOrBundle: "test"}
+	cmdtesting.InitCommand(command, []string{"--format", "yaml"})
+	ctx := commandContextForTest(c)
+	err := command.Run(ctx)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(cmdtesting.Stdout(ctx), gc.Equals, `
+type: charm
+id: charmCHARMcharmCHARMcharmCHARM01
+name: wordpress
+description: This will install and setup WordPress optimized to run in the cloud.
+publisher: Wordress Charmers
+summary: WordPress is a full featured web blogging tool, this charm deploys it.
+series:
+- bionic
+- xenial
+storeurl: https://someurl.com/wordpress
+tags:
+- app
+- seven
+charm:
+  config:
+    options:
```

Comment:
maybe we don't need a --config TODO with this change.

Reply:
Agreed